R/conditions.R

```diff
@@ -232,6 +232,15 @@ cnd_type_message <- function(x,
   if (is_null(types)) {
     x_type <- vec_ptype_full(x)
     y_type <- vec_ptype_full(y)
+
+    if (identical(x_type, y_type)) {
```

Comment:
I know we don't have a great method for determining when to add this extra message, but I'm a bit worried about using `vec_ptype_full()`. Some types use attribute data when constructing the full ptype string, which would result in this message not being applied.

```r
> vctrs::vec_ptype_full(new_datetime(tzone = "America/New_York"))
[1] "datetime<America/New_York>"
> vctrs::vec_ptype_full(new_datetime(tzone = "America/Pacific"))
[1] "datetime<America/Pacific>"
```

What if we just strip all attributes and compare the underlying raw data?

Reply:
Actually I think I meant to check for `class(x)[[1]]` originally, to determine whether to show the warning.

But now I realise the whole thing is not well founded. My reasoning was that we would only end up here if the is-same-class fallback failed, but we could end up here from an actual ptype2 method if the common type determination failed.

Reply:
We now check for `class()` instead of the type label. And we detect when we are called from dispatch. In that case we don't display we don't display the attributes advice.